### ISSUE TITLE
Selenium runner fails to start on ubuntu-latest after switching to Ubuntu 22

### ISSUE THREAD

**Issue by mvlasovatl** [actions/runner-images#6714]
### Description

Hello team!
The app I support has Selenium integration tests that start Firefox and run there. They were working fine on Nov 25th and today on Dec 7th I discovered, that most of them started failing on the same commit without changes from my side. I think it is related to the switch from Ubuntu 20 to Ubuntu 22 for `ubuntu-latest` runner described here: https://github.blog/changelog/2022-12-01-github-actions-larger-runners-using-ubuntu-latest-label-will-now-use-ubuntu-22-04/ . Downgrading runner to `ubuntu-20.04` fixes the issues. Could you please fix Ubuntu 22 image to allow Selenium tests run there again?

Successful build on `dev` branch (Nov 25th, Ubuntu 20): https://github.com/atlassian-labs/atlassian-slack-integration-server/actions/runs/3549057850/jobs/5961116540
Failed build on `dev` branch (Dec 7th, Ubuntu 22): https://github.com/atlassian-labs/atlassian-slack-integration-server/actions/runs/3628712902/jobs/6145292063
Stacktrace:
```
Build info: version: '3.141.59-atlassian-1', revision: '8fea9e9515', time: '2021-03-31T05:00:23'
System info: host: 'fv-az401-608', ip: '10.1.1.95', os.name: 'Linux', os.arch: 'amd64', os.version: '5.15.0-1023-azure', java.version: '1.8.0_352'
Driver info: driver.version: FirefoxDriver
remote stacktrace: 
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.openqa.selenium.remote.W3CHandshakeResponse.lambda$errorHandler$0(W3CHandshakeResponse.java:62)
	at org.openqa.selenium.remote.HandshakeResponse.lambda$getResponseFunction$0(HandshakeResponse.java:30)
	at org.openqa.selenium.remote.ProtocolHandshake.lambda$createSession$0(ProtocolHandshake.java:126)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:958)
	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:126)
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:499)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:486)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:152)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:531)
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:128)
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:74)
	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:136)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:83)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:552)
	at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:213)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:131)
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:162)
	at com.atlassian.webdriver.browsers.firefox.FirefoxBrowser.constructFirefoxDriver(FirefoxBrowser.java:202)
	at com.atlassian.webdriver.browsers.firefox.FirefoxBrowser.getFirefoxDriver(FirefoxBrowser.java:85)
	at com.atlassian.webdriver.WebDriverFactory.getDriverNoRetry(WebDriverFactory.java:110)
	at com.atlassian.webdriver.WebDriverFactory.lambda$getDriverContext$0(WebDriverFactory.java:73)
	at com.atlassian.fugue.retry.RetrySupplier.get(RetrySupplier.java:93)
	at com.atlassian.webdriver.WebDriverFactory.getDriverContext(WebDriverFactory.java:78)
	at com.atlassian.webdriver.LifecycleAwareWebDriverGrid.getDriverContext(LifecycleAwareWebDriverGrid.java:89)
	at com.atlassian.webdriver.pageobjects.DefaultWebDriverTester.<init>(DefaultWebDriverTester.java:22)
	at com.atlassian.jira.pageobjects.JiraTestedProduct.<init>(JiraTestedProduct.java:96)
	at com.atlassian.jira.pageobjects.JiraTestedProduct.<init>(JiraTestedProduct.java:174)
	at com.atlassian.jira.pageobjects.setup.SingleJiraWebTestRunner.newLocalTestedProduct(SingleJiraWebTestRunner.java:36)
	at com.atlassian.jira.pageobjects.setup.SingleJiraWebTestRunner.access$000(SingleJiraWebTestRunner.java:18)
	at com.atlassian.jira.pageobjects.setup.SingleJiraWebTestRunner$1.create(SingleJiraWebTestRunner.java:27)
	at com.atlassian.jira.pageobjects.setup.SingleJiraWebTestRunner$1.create(SingleJiraWebTestRunner.java:24)
	at io.atlassian.util.concurrent.LazyReference$Sync.run(LazyReference.java:332)
	at io.atlassian.util.concurrent.LazyReference.getInterruptibly(LazyReference.java:150)
	at io.atlassian.util.concurrent.LazyReference.get(LazyReference.java:116)
	at com.atlassian.jira.pageobjects.setup.SingleJiraWebTestRunner.<init>(SingleJiraWebTestRunner.java:32)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.junit.internal.builders.AnnotatedBuilder.buildRunner(AnnotatedBuilder.java:104)
	at org.junit.vintage.engine.discovery.DefensiveAllDefaultPossibilitiesBuilder$DefensiveAnnotatedBuilder.buildRunner(DefensiveAllDefaultPossibilitiesBuilder.java:114)
	at org.junit.internal.builders.AnnotatedBuilder.runnerForClass(AnnotatedBuilder.java:86)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:70)
	at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:37)
	at org.junit.vintage.engine.discovery.DefensiveAllDefaultPossibilitiesBuilder.runnerForClass(DefensiveAllDefaultPossibilitiesBuilder.java:57)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:70)
	at org.junit.vintage.engine.discovery.ClassSelectorResolver.resolveTestClass(ClassSelectorResolver.java:66)
	at org.junit.vintage.engine.discovery.ClassSelectorResolver.resolve(ClassSelectorResolver.java:47)
	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.lambda$resolve$2(EngineDiscoveryRequestResolution.java:134)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1361)
	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:126)
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:499)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:486)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:152)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:531)
	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolve(EngineDiscoveryRequestResolution.java:185)
	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolve(EngineDiscoveryRequestResolution.java:125)
	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolveCompletely(EngineDiscoveryRequestResolution.java:91)
	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.run(EngineDiscoveryRequestResolution.java:82)
	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolver.resolve(EngineDiscoveryRequestResolver.java:113)
	at org.junit.vintage.engine.discovery.VintageDiscoverer.discover(VintageDiscoverer.java:44)
	at org.junit.vintage.engine.VintageTestEngine.discover(VintageTestEngine.java:63)
	at org.junit.platform.launcher.core.DefaultLauncher.discoverEngineRoot(DefaultLauncher.java:168)
	at org.junit.platform.launcher.core.DefaultLauncher.discoverRoot(DefaultLauncher.java:155)
	at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:120)
	at org.apache.maven.surefire.junitplatform.TestPlanScannerFilter.accept(TestPlanScannerFilter.java:56)
	at org.apache.maven.surefire.util.DefaultScanResult.applyFilter(DefaultScanResult.java:102)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.scanClasspath(JUnitPlatformProvider.java:129)
	... 5 more
```

### Platforms affected

- [ ] Azure DevOps
- [X] GitHub Actions - Standard Runners
- [ ] GitHub Actions - Larger Runners

### Runner images affected

- [ ] Ubuntu 18.04
- [ ] Ubuntu 20.04
- [X] Ubuntu 22.04
- [ ] macOS 10.15
- [ ] macOS 11
- [ ] macOS 12
- [ ] Windows Server 2019
- [ ] Windows Server 2022

### Image version and build link

Image: ubuntu-22.04
Version: 20221127.1
Failed build link: https://github.com/atlassian-labs/atlassian-slack-integration-server/actions/runs/3628712902/jobs/6145292063

### Is it regression?

https://github.com/atlassian-labs/atlassian-slack-integration-server/actions/runs/3549057850/jobs/5961116540

### Expected behavior

I want Selenium integration tests passing on `ubuntu-latest`  image (Ubuntu 22 since Dec 1st).

### Actual behavior

The following stacktrace is printed, the tests fail.
```
Build info: version: '3.141.59-atlassian-1', revision: '8fea9e9515', time: '2021-03-31T05:00:23'
System info: host: 'fv-az401-608', ip: '10.1.1.95', os.name: 'Linux', os.arch: 'amd64', os.version: '5.15.0-1023-azure', java.version: '1.8.0_352'
Driver info: driver.version: FirefoxDriver
remote stacktrace: 
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.openqa.selenium.remote.W3CHandshakeResponse.lambda$errorHandler$0(W3CHandshakeResponse.java:62)
	at org.openqa.selenium.remote.HandshakeResponse.lambda$getResponseFunction$0(HandshakeResponse.java:30)
	at org.openqa.selenium.remote.ProtocolHandshake.lambda$createSession$0(ProtocolHandshake.java:126)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.Spliterators$ArraySpliterator.tryAdvance(Spliterators.java:958)
	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:126)
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:499)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:486)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:152)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:531)
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:128)
	at org.openqa.selenium.remote.ProtocolHandshake.createSession(ProtocolHandshake.java:74)
	at org.openqa.selenium.remote.HttpCommandExecutor.execute(HttpCommandExecutor.java:136)
	at org.openqa.selenium.remote.service.DriverCommandExecutor.execute(DriverCommandExecutor.java:83)
	at org.openqa.selenium.remote.RemoteWebDriver.execute(RemoteWebDriver.java:552)
	at org.openqa.selenium.remote.RemoteWebDriver.startSession(RemoteWebDriver.java:213)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:131)
	at org.openqa.selenium.firefox.FirefoxDriver.<init>(FirefoxDriver.java:162)
	at com.atlassian.webdriver.browsers.firefox.FirefoxBrowser.constructFirefoxDriver(FirefoxBrowser.java:202)
	at com.atlassian.webdriver.browsers.firefox.FirefoxBrowser.getFirefoxDriver(FirefoxBrowser.java:85)
	at com.atlassian.webdriver.WebDriverFactory.getDriverNoRetry(WebDriverFactory.java:110)
	at com.atlassian.webdriver.WebDriverFactory.lambda$getDriverContext$0(WebDriverFactory.java:73)
	at com.atlassian.fugue.retry.RetrySupplier.get(RetrySupplier.java:93)
	at com.atlassian.webdriver.WebDriverFactory.getDriverContext(WebDriverFactory.java:78)
	at com.atlassian.webdriver.LifecycleAwareWebDriverGrid.getDriverContext(LifecycleAwareWebDriverGrid.java:89)
	at com.atlassian.webdriver.pageobjects.DefaultWebDriverTester.<init>(DefaultWebDriverTester.java:22)
	at com.atlassian.jira.pageobjects.JiraTestedProduct.<init>(JiraTestedProduct.java:96)
	at com.atlassian.jira.pageobjects.JiraTestedProduct.<init>(JiraTestedProduct.java:174)
	at com.atlassian.jira.pageobjects.setup.SingleJiraWebTestRunner.newLocalTestedProduct(SingleJiraWebTestRunner.java:36)
	at com.atlassian.jira.pageobjects.setup.SingleJiraWebTestRunner.access$000(SingleJiraWebTestRunner.java:18)
	at com.atlassian.jira.pageobjects.setup.SingleJiraWebTestRunner$1.create(SingleJiraWebTestRunner.java:27)
	at com.atlassian.jira.pageobjects.setup.SingleJiraWebTestRunner$1.create(SingleJiraWebTestRunner.java:24)
	at io.atlassian.util.concurrent.LazyReference$Sync.run(LazyReference.java:332)
	at io.atlassian.util.concurrent.LazyReference.getInterruptibly(LazyReference.java:150)
	at io.atlassian.util.concurrent.LazyReference.get(LazyReference.java:116)
	at com.atlassian.jira.pageobjects.setup.SingleJiraWebTestRunner.<init>(SingleJiraWebTestRunner.java:32)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)
	at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62)
	at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45)
	at java.lang.reflect.Constructor.newInstance(Constructor.java:423)
	at org.junit.internal.builders.AnnotatedBuilder.buildRunner(AnnotatedBuilder.java:104)
	at org.junit.vintage.engine.discovery.DefensiveAllDefaultPossibilitiesBuilder$DefensiveAnnotatedBuilder.buildRunner(DefensiveAllDefaultPossibilitiesBuilder.java:114)
	at org.junit.internal.builders.AnnotatedBuilder.runnerForClass(AnnotatedBuilder.java:86)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:70)
	at org.junit.internal.builders.AllDefaultPossibilitiesBuilder.runnerForClass(AllDefaultPossibilitiesBuilder.java:37)
	at org.junit.vintage.engine.discovery.DefensiveAllDefaultPossibilitiesBuilder.runnerForClass(DefensiveAllDefaultPossibilitiesBuilder.java:57)
	at org.junit.runners.model.RunnerBuilder.safeRunnerForClass(RunnerBuilder.java:70)
	at org.junit.vintage.engine.discovery.ClassSelectorResolver.resolveTestClass(ClassSelectorResolver.java:66)
	at org.junit.vintage.engine.discovery.ClassSelectorResolver.resolve(ClassSelectorResolver.java:47)
	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.lambda$resolve$2(EngineDiscoveryRequestResolution.java:134)
	at java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:193)
	at java.util.ArrayList$ArrayListSpliterator.tryAdvance(ArrayList.java:1361)
	at java.util.stream.ReferencePipeline.forEachWithCancel(ReferencePipeline.java:126)
	at java.util.stream.AbstractPipeline.copyIntoWithCancel(AbstractPipeline.java:499)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:486)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:472)
	at java.util.stream.FindOps$FindOp.evaluateSequential(FindOps.java:152)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.findFirst(ReferencePipeline.java:531)
	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolve(EngineDiscoveryRequestResolution.java:185)
	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolve(EngineDiscoveryRequestResolution.java:125)
	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.resolveCompletely(EngineDiscoveryRequestResolution.java:91)
	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolution.run(EngineDiscoveryRequestResolution.java:82)
	at org.junit.platform.engine.support.discovery.EngineDiscoveryRequestResolver.resolve(EngineDiscoveryRequestResolver.java:113)
	at org.junit.vintage.engine.discovery.VintageDiscoverer.discover(VintageDiscoverer.java:44)
	at org.junit.vintage.engine.VintageTestEngine.discover(VintageTestEngine.java:63)
	at org.junit.platform.launcher.core.DefaultLauncher.discoverEngineRoot(DefaultLauncher.java:168)
	at org.junit.platform.launcher.core.DefaultLauncher.discoverRoot(DefaultLauncher.java:155)
	at org.junit.platform.launcher.core.DefaultLauncher.discover(DefaultLauncher.java:120)
	at org.apache.maven.surefire.junitplatform.TestPlanScannerFilter.accept(TestPlanScannerFilter.java:56)
	at org.apache.maven.surefire.util.DefaultScanResult.applyFilter(DefaultScanResult.java:102)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.scanClasspath(JUnitPlatformProvider.java:129)
	... 5 more
```

### Repro steps

1. For the repository https://github.com/atlassian-labs/atlassian-slack-integration-server
2. Go to Actions tab.
3. Select "All Unit and Integration Tests" workflow.
4. Click "Run workflow" button to trigger the build. In *Use workflow from* and *Branch/tag/hash to use (defaults to master)* fields select `ubuntu-latest` and click "Run workflow" button.
5. See the build results.

Expected: all jobs pass.
Actual: Unit tests jobs pass, all integration tests except ones for Jira 8.9.0 fail.
<img width="358" alt="ActionsFailedJobs" src="https://user-images.githubusercontent.com/66078253/206245579-2d0e40a1-709d-4c98-8a3e-69b06d534e4b.png">

**Comment by erik-bershel** [comment]
@mvlasovatl Ubuntu22 and Ubuntu20 images have a bit different list of installed packages due to difference in list of supported software both for our and for Canonical repos. Are you already checked that all the packages used in your tests dependencies are already on board for Ubuntu22?

**Comment by mvlasovatl** [comment]
Hey @erik-bershel ! To be honest, I don't know how to check what dependencies are needed to run integration tests. From pre-installed software just Maven and `bash` are used. Java is installed by a separate action, Java dependencies are loaded by Maven from net of cache every time.

Some native libraries for running integration tests are installed by the following script using `apt-get` that is run by every job: https://github.com/atlassian-labs/atlassian-slack-integration-server/blob/master/bin/build/install-int-test-libs.sh . I got that list after trying to run tests in the past. That time I saw a lot of messages regarding missing native library. I added that libraries to the script to install them before running tests and it helped. In the build logs from Dec 7th I don't find any messages regarding any other native library missing. All I get is that long stacktrace.

**Comment by mikhailkoliada** [comment]
@mvlasovatl Hello!

We generally can't guarantee that your workflows will work after the version of `ubuntu-latest` got changed. From the investigation I can say that we have not identified anything could be fixed in the runner-images repo, if `ubuntu-20.04` is still working for you, I'd encourage you hardcode it in your pipeline.

If you have questions left, feel free to reach us again and ask :)
